pkg/schedule/operator/create_operator.go

```diff
@@ -80,10 +80,9 @@ func CreateRemovePeerOperator(desc string, ci sche.SharedCluster, kind OpKind, r
 }
 
 // CreateTransferLeaderOperator creates an operator that transfers the leader from a source store to a target store.
-func CreateTransferLeaderOperator(desc string, ci sche.SharedCluster, region *core.RegionInfo, targetStoreID uint64, targetStoreIDs []uint64, kind OpKind) (*Operator, error) {
+func CreateTransferLeaderOperator(desc string, ci sche.SharedCluster, region *core.RegionInfo, targetStoreID uint64, kind OpKind) (*Operator, error) {
 	return NewBuilder(desc, ci, region, SkipOriginJointStateCheck).
 		SetLeader(targetStoreID).
-		SetLeaders(targetStoreIDs).
```

Comment:
Why remove it?

Reply:
We do not need `targetStoreIDs`, which is used in `evict_leader.go` before.

Reply:
It's an optimization, why we don't need it?

Reply:
`SetLeaders` checks learner and unhealthy  peer. https://github.com/tikv/pd/blob/master/pkg%2Fschedule%2Foperator%2Fbuilder.go#L301-L301

`SetLeader` also check these. https://github.com/tikv/pd/blob/master/pkg%2Fschedule%2Foperator%2Fbuilder.go#L284-L284

SetLeaders sort target stores according to store id. This PR sort target stores according to score.

Reply:
targetLeaderStoreIDs is used previously, but removed by this PR?

Reply:
Yes, we use it in evict leader scheduler previously to select targets. It is replaced with this pr.

Reply:
We allow multiple targets in op step before, this PR changes it.

Reply:
It only used in evict leader, after this pr there is no other scheduler using it. So I remove it. 
If we will use it in the future, I will recover it.

Reply:
It was O(n) previously and it is O(nlogn) now. 
Considering that the number of sorts is usually two (three replicas) or four (five replicas), the difference is not too big.

Reply:
See https://github.com/tikv/tikv/issues/10602